pymeasure/instruments/keithley/keithleyDAQ6510.py

```diff
@@ -0,0 +1,103 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2024 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+from pymeasure.instruments import Instrument
+from pymeasure.instruments.generic_types import SCPIMixin
+from .buffer import KeithleyBuffer
+
+
+class KeithleyDAQ6510(KeithleyBuffer, SCPIMixin, Instrument):
+    """ Represents the Keithley DAQ6510 Data Acquisition Logging Multimeter System
+    and provides a high-level interface for interacting with the instrument.
+
+    .. code-block:: python
+
+        keithley = KeithleyDAQ6510("GPIB::1")
+        keithley = KeithleyDAQ6510("TCPIP::192.168.1.1::INSTR")
+
+    """
+
+    def __init__(self, adapter, name="Keithley DAQ6510", **kwargs):
+        super().__init__(
+            adapter,
+            name,
+            **kwargs
+        )
+
+    open_channel = Instrument.setting(
+        ":ROUT:OPEN (@%d)",
+        """Set a single channel to open."""
+    )
+
+    close_channel = Instrument.setting(
+        ":ROUT:CLOS (@%d)",
+        """Set a single channel to closed."""
+    )
```

Comment:
You use it as an "action word", therefore these properties should be methods instead.

Reply:
Changed to be methods